front/app/containers/Admin/pagesAndMenu/routes.tsx

```diff
@@ -75,5 +76,9 @@ export default () => ({
       element: <EditHomepage />,
     },
     ...moduleConfiguration.routes['admin.pages-menu'],
+    {
+      path: 'pages/edit/:pageId', // used only if the same route is not loaded by `customizable_navbar` module
```

Comment:
Ideally, we could use this route not only if the module is loaded but also if the feature is disabled. In this case, we wouldn't need any changes in `front/app/modules/commercial/customizable_navbar/admin/containers/EditPageForm/index.tsx`. But I'm not sure how to do it.

Also, probably, it will be removed/changed/refactored in i2 again.

Reply:
I don't think there's any way to do that from within the module, if it's not loaded you don't have access to any of this. I think this is a decent compromise, and I agree this will be revamped in i2 (or soon after) so this seems fine

Reply:
You can do this via a module, in the same way we do this for inserting tabs, etc.

You bring fixed data (the array with routes here) to the state, then have an Outlet that inserts data into that state when the module has mounted. You could add that feature flag condition in the module to this as well.

Always the same pattern. I'll do a session on this maybe next week in the CDM, because it keeps confusing people so it seems we haven't explained it well. :-)

Reply:
PS, not sure what you mean with "used only if the same route is not loaded by `customizable_navbar` module". 

Have you tested the right component is there if the module does load? Because you're adding this in the core without any conditions, which means it'll always be added to the routes AFAIK. 🤔 do they get filtered somewhere for duplicates?

I need to understand better what you're trying to achieve here to judge if this is a good solution. Could you describe the problem (bug) and how you're trying to solve it?

Reply:
(Always useful if you explain in a comment here on GitHub what you're code is doing and some context, makes it easier to review.)

Reply:
> You can do this via a module, in the same way we do this for inserting tabs, etc.

But there we use our `insertConfiguration` function. And we don't use it with routes, do we?

Reply:
> But there we use our insertConfiguration function. And we don't use it with routes, do we?

Not at the moment. We could look into it if would make our lives easier.

Reply:
In any case, if you're adding the route in the core, there's no need to load it in the module anymore.

Reply:
> Not at the moment. We could look into it if would make our lives easier.

Maybe not worth it as anyway we're likely going to change it soon?

> if you're adding the route in the core, there's no need to load it in the module anymore

But the `element:` part is different in two cases (when the module is loaded or not). We could use an outlet and make a condition in the core like

```ts
const customizableNavbarEnabled = useFeatureFlag({ name: 'customizable_navbar' });
customizableNavbarEnabled ? (
    <Outlet ...>
  ) : (
    <EditPageFormNotInNavbar hideSlugInput={!customizableNavbarEnabled} />
```
But I suppose you would be against it. Also, we would need to handle the `isPageInNavbar` condition somehow.

Reply:
> Maybe not worth it as anyway we're likely going to change it soon?

I didn't mean to do it now indeed.

> But the element: part is different in two cases (when the module is loaded or not). We could use an outlet and make a condition in the core like

Outlets should not render conditionally. Outlets need to always render, conditions need to go inside modules where you determine what to render via the Outlet.

Reply:
> Outlets should not render conditionally.

Yes, sure. Then rendering only `EditPageFormNotInNavbar` by condition and always rendering the outlet. It's very similar.